docs/how_to/use_optimizationlevel.rst

```diff
@@ -0,0 +1,278 @@
+the internals of the :mod:`~.transpiler` in: :ref:`working_with_preset_pass_managers`.
+
+Initialize the quantum circuit
+==============================
+
+The effect of setting the ``optimization_level`` will differ depending on the backend you are using.
+For this example, you will explore the `CSWAP <https://qiskit.org/documentation/stubs/qiskit.circuit.QuantumCircuit.cswap.html>`_ gate,
+which is a three qubit gate which will be transpiled into one and two qubit gates.
+
+.. plot::
+    :include-source:
+
+    from qiskit import QuantumCircuit 
+    from qiskit.compiler import transpile
+    from qiskit.providers.fake_provider import FakeQuito
+
+    backend = FakeQuito()
+
+    qc = QuantumCircuit(3) # Initialize the quantum circuit with 3 qubits.
+    
+    qc.cswap(0,1,2) # Add the cswap gate to the quantum circuit.
+    
+    qc.draw("mpl")
+
+
+Using backend’s information
+===========================
+
+
+You should adhere to the specific configuration of your backend when utilizing :meth:`~qiskit.transpile` . 
+This process entails breaking down your circuit into :attr:`~qiskit.transpile.basis_gates` and considering the physical connections specified in the 
+:attr:`~qiskit.transpile.coupling_map` for two qubit gates.
+Given the presence of noise in the backend, it is crucial to optimize your circuit by adjusting the ``optimization_level`` parameter. 
+This will help minimize the number of circuit operations and enhance the overall performance.
+
+What each optimization level does
+---------------------------------
+
+
+When using a backend, you can access its properties through the instruction  :meth:`backend.configuration()`.
+These properties, such as basis gates, coupling map, and init layout, play a crucial role in shaping the behavior of the quantum circuit.
+
+For example, with :meth:`~qiskit.providers.fake_provider.FakeQuito`, you can learn about its qubit connections and the gates it uses to generate your quantum circuits.
+
+.. testcode::
+
+    print(f"Basis gates of your backend: {backend.configuration().basis_gates}")
+    print(f"Coupling map of your backend: ",{backend.configuration().coupling_map}")
+
+.. testoutput::
+
+    Basis gates of your backend:  ['id', 'rz', 'sx', 'x', 'cx', 'reset']
+    Coupling map of your backend:  [[0, 1], [1, 0], [1, 2], [1, 3], [2, 1], [3, 1], [3, 4], [4, 3]]
+
+When setting the ``optimization_level`` to 0, the resulting quantum circuit is not optimized and simply mapped to the device, considering a trivial layout and stochastic swap. 
+The coupling map, represented by the subset ``[[0,1],[1,0],[1,2],[2,1]]``, indicates the physical qubits available in the backend. 
+In this configuration, the quantum circuit is transformed into a combination of one and two-qubit gates,
+represented by the ``['id', 'rz', 'sx', 'x', 'cx', 'reset']``.
+
+.. testcode::
+
+    qc_b0 = transpile(qc,backend=backend,optimization_level=0)
+    qc_b0.draw("mpl")                          
```

Comment:
Is this leftover? It sounds like you only want to use plot now. Ditto to all the other testcode directives in this file.

Reply:
You can use the `:include-source:` option: https://matplotlib.org/stable/api/sphinxext_plot_directive_api.html#options on the plot directive to show the source code. in the single block with the visualization. For example, you can see this used already here: https://github.com/Qiskit/qiskit-terra/blob/df2ddcadf3c6d59b8bab8b0b1ceff14234fa82b5/qiskit/circuit/operation.py#L32